"use strict";(self.webpackChunkmspc=self.webpackChunkmspc||[]).push([[386],{5895:(e,n,t)=>{t.r(n),t.d(n,{assets:()=>c,contentTitle:()=>o,default:()=>p,frontMatter:()=>r,metadata:()=>s,toc:()=>l});const s=JSON.parse('{"id":"welcome","title":"Welcome","description":"The analysis of ChIP-seq samples outputs a number","source":"@site/docs/welcome.md","sourceDirName":".","slug":"/","permalink":"/MSPC/docs/","draft":false,"unlisted":false,"editUrl":"https://github.com/Genometric/MSPC/tree/dev/website/docs/welcome.md","tags":[],"version":"current","frontMatter":{"title":"Welcome","slug":"/"},"sidebar":"someSidebar","next":{"title":"Quick Start","permalink":"/MSPC/docs/quick_start"}}');var i=t(4848),a=t(8453);const r={title:"Welcome",slug:"/"},o="About",c={},l=[];function h(e){const n={a:"a",h1:"h1",header:"header",li:"li",p:"p",strong:"strong",ul:"ul",...(0,a.R)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsx)(n.p,{children:'The analysis of ChIP-seq samples outputs a number\nof enriched regions (commonly known as "peaks"),\neach indicating a protein-DNA interaction or a\nspecific chromatin modification. When replicate\nsamples are analyzed, overlapping peaks are expected.\nThis repeated evidence can therefore be used to\nlocally lower the minimum significance required to\naccept a peak. Here, we propose a method for joint\nanalysis of weak peaks. Given a set of peaks from\n(biological or technical) replicates, the method\ncombines the p-values of overlapping enriched regions,\nand allows the "rescue" of weak peaks occurring in\nmore than one replicate.'}),"\n",(0,i.jsx)(n.p,{children:"MSPC comparatively evaluates ChIP-seq peaks and\ncombines the statistical significance of\nrepeated evidences, with the aim of lowering the\nminimum significance required to \u201crescue\u201d\nweak peaks; hence reducing false-negatives."}),"\n",(0,i.jsx)(n.p,{children:"MSPC can be used from:"}),"\n",(0,i.jsxs)(n.ul,{children:["\n",(0,i.jsxs)(n.li,{children:["\n",(0,i.jsxs)(n.p,{children:[(0,i.jsx)(n.a,{href:"/MSPC/docs/cli/about",children:(0,i.jsx)(n.strong,{children:"Command Line Interface (CLI)"})}),": call MSPC CLI from your favorite terminal with necessary\narguments, and it writes the analysis results to BED files."]}),"\n"]}),"\n",(0,i.jsxs)(n.li,{children:["\n",(0,i.jsxs)(n.p,{children:[(0,i.jsx)(n.a,{href:"/MSPC/docs/library/install",children:(0,i.jsx)(n.strong,{children:"As a library in your project"})}),": call it from your program, and it returns analysis\nresults to your program."]}),"\n"]}),"\n",(0,i.jsxs)(n.li,{children:["\n",(0,i.jsxs)(n.p,{children:[(0,i.jsx)(n.a,{href:"https://bioconductor.org/packages/release/bioc/html/rmspc.html",children:(0,i.jsx)(n.strong,{children:"As an R package from Bioconductor"})}),":\nThe Bioconductor package is a wrapper around the CLI application, hence\noffering the same functionality as the CLI application. The package can be\ninvoked from R to process files or\n",(0,i.jsx)(n.a,{href:"https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html",children:"GenomicRanges"}),"\n(Granges) objects, and the results are stored as Granges objects.\nPlease refer to ",(0,i.jsx)(n.a,{href:"https://bioconductor.org/packages/release/bioc/vignettes/rmspc/inst/doc/rmpsc.html",children:"Bioconductor user guide"}),"."]}),"\n"]}),"\n"]}),"\n",(0,i.jsx)(n.header,{children:(0,i.jsx)(n.h1,{id:"about",children:"About"})}),"\n",(0,i.jsxs)(n.p,{children:["With most peak callers (e.g, MACS), the false-positive\nrate is a function of a user-defined p-value threshold,\nwhere the more conservative thresholds result in lower\nfalse-positive rates---the penalty of which is the\nincrease in the number of false-negatives. While several\nprobabilistic methods are developed to jointly model\nbinding affinities across replicated samples to identify\ncombinatorial enrichment patterns (e.g.,\n",(0,i.jsx)(n.a,{href:"https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38",children:"jMOSAiCS"}),",\nor ",(0,i.jsx)(n.a,{href:"https://academic.oup.com/biostatistics/article/15/2/296/226404",children:"this"}),",\nor ",(0,i.jsx)(n.a,{href:"https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14",children:"this"}),",\nor ",(0,i.jsx)(n.a,{href:"https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full",children:"this"}),"),\nthe more commonly used peak callers (e.g., MACS) operate\non single samples. Additionally, the choice of peak\ncaller is generally an established step in a ChIP-seq\nanalysis pipeline, since altering it may require changes\nto the data pre- and post-processing of the pipeline.\nTherefore, the remainder of this manuscript is focused\non the post peak calling methods to lower false-positive\nrates as they can be applied on the output of any peak\ncaller with minimal changes to the established analysis\npipeline, and can combine evidence across replicated samples."]}),"\n",(0,i.jsx)(n.h1,{id:"motivation",children:"Motivation"}),"\n",(0,i.jsxs)(n.p,{children:["Ultimately, with most peak callers (e.g, MACS), the false-positive\nrate is a function of a user-defined p-value threshold, where the more\nconservative thresholds result in lower false-positive rates---the penalty\nof which is the increase in the number of false-negatives. While several\nprobabilistic methods are developed to jointly model binding affinities\nacross replicated samples to identify combinatorial enrichment patterns\n(e.g.,\n",(0,i.jsx)(n.a,{href:"https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38",children:"REF1"}),",\n",(0,i.jsx)(n.a,{href:"https://academic.oup.com/biostatistics/article/15/2/296/226404",children:"REF2"}),",\n",(0,i.jsx)(n.a,{href:"https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14",children:"REF3"}),",\n",(0,i.jsx)(n.a,{href:"https://academic.oup.com/bioinformatics/article/31/1/17/2366199",children:"REF4"}),",\n",(0,i.jsx)(n.a,{href:"https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full",children:"REF5"}),"),\nthe more commonly used peak callers\n(e.g., MACS) operate on single samples. Additionally, the choice of peak\ncaller is generally an established step in a ChIP-seq analysis pipeline,\nsince altering it may require changes to the data pre- and post-processing\nof the pipeline."]})]})}function p(e={}){const{wrapper:n}={...(0,a.R)(),...e.components};return n?(0,i.jsx)(n,{...e,children:(0,i.jsx)(h,{...e})}):h(e)}},8453:(e,n,t)=>{t.d(n,{R:()=>r,x:()=>o});var s=t(6540);const i={},a=s.createContext(i);function r(e){const n=s.useContext(a);return s.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function o(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(i):e.components||i:r(e.components),s.createElement(a.Provider,{value:n},e.children)}}}]);