(window.webpackJsonp=window.webpackJsonp||[]).push([[11],{107:function(e,t,n){"use strict";n.r(t),t.default=n.p+"assets/images/sets-b9292602d22d0c4afcb84f256dfcf274.svg"},162:function(e,t,n){"use strict";n.r(t),t.default=n.p+"assets/images/simplified_flow_chart-3e5142cb12287ebbcbb1efa2ae0c7ae8.svg"},82:function(e,t,n){"use strict";n.r(t),n.d(t,"frontMatter",(function(){return o})),n.d(t,"metadata",(function(){return s})),n.d(t,"rightToc",(function(){return c})),n.d(t,"default",(function(){return p}));var a=n(3),r=n(7),i=(n(0),n(95)),o={title:"About"},s={unversionedId:"method/about",id:"method/about",isDocsHomePage:!1,title:"About",description:"The analysis of ChIP-seq samples outputs a number of enriched regions",source:"@site/docs/method/about.md",slug:"/method/about",permalink:"/MSPC/docs/method/about",editUrl:"https://github.com/Genometric/MSPC/tree/dev/website/docs/method/about.md",version:"current",sidebar:"someSidebar",previous:{title:"Sample Data",permalink:"/MSPC/docs/sample_data"},next:{title:"Sets",permalink:"/MSPC/docs/method/sets"}},c=[{value:"Remarks",id:"remarks",children:[]}],l={rightToc:c};function p(e){var t=e.components,o=Object(r.a)(e,["components"]);return Object(i.b)("wrapper",Object(a.a)({},l,o,{components:t,mdxType:"MDXLayout"}),Object(i.b)("p",null,'The analysis of ChIP-seq samples outputs a number of enriched regions\n(commonly known as "peaks"), each indicating a protein-DNA interaction\nor a specific chromatin modification.'),Object(i.b)("p",null,"When replicate samples are analyzed, overlapping peaks are expected.\nThis repeated evidence can therefore be used to locally lower the\nminimum significance required to accept a peak. MSPC is a method for\njoint analysis of peaks."),Object(i.b)("p",null,'Given a set of peaks from (biological or technical) replicates,\nMSPC combines the p-values of overlapping enriched regions, and allows\nthe "rescue" of weak peaks occurring in more than one replicate.'),Object(i.b)("h2",{id:"remarks"},"Remarks"),Object(i.b)("p",null,'MSPC rigorously combines the statistical significance of the overlapping\nenriched regions (ER), in order to "rescue" weak peaks, which would probably\nbe discarded in a single-sample analysis, when their combined evidence\nacross multiple samples is sufficiently strong.\n[',Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://doi.org/10.1093/bioinformatics/btv293"}),"Ref"),"]"),Object(i.b)("p",null,'MSPC takes as input, for each replicate, a list of ERs and a measure of\ntheir individual significance in terms of a p-value.\nStarting from a permissive call, it divides the initial ERs in "stringent"\n(highly significant) and "weak" (moderately significant), and it assesses\nthe presence of overlapping enriched regions across multiple replicates.\nNon-overlapping regions can be penalized or discarded according to specific\nneeds. '),Object(i.b)("p",null,"The significance of the overlapping regions is rigorously combined with the\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://en.wikipedia.org/wiki/Fisher%27s_method"}),"Fisher's method")," to obtain\na global score. This score is assessed against an adjustable threshold on\nthe combined evidence, and peaks in each replicate are either confirmed or\ndiscarded."),Object(i.b)("p",null,"Finally, in order to account for multiple testing correction, MSPC applies the\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://en.wikipedia.org/wiki/False_discovery_rate#Benjamini%E2%80%93Hochberg_procedure"}),"Benjamini-Hochberg procedure"),",\nand outputs ERs with false-discovery rate smaller than an adjustable threshold. "),Object(i.b)("p",null,"This flow is captured in the following flowchart for each ER of each replicate: "),Object(i.b)("p",null,Object(i.b)("img",{alt:"Simplified Flow Chart",src:n(162).default})),Object(i.b)("p",null,"(This flowchart is a simplified version of the flowchart available in\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://doi.org/10.1093/bioinformatics/btv293"}),"MSPC's manuscript"),".)"),Object(i.b)("p",null,"In other words:"),Object(i.b)("ul",null,Object(i.b)("li",{parentName:"ul"},"for each replicate, MSPC classifies input ERs as ",Object(i.b)("em",{parentName:"li"},"stringent"),", ",Object(i.b)("em",{parentName:"li"},"weak"),",\nand ",Object(i.b)("em",{parentName:"li"},"background"),", based on their p-value;"),Object(i.b)("li",{parentName:"ul"},"performs a comparative analysis, and based on the combined stringency test,\nit classifies ",Object(i.b)("em",{parentName:"li"},"stringent")," and ",Object(i.b)("em",{parentName:"li"},"weak")," ERs as ",Object(i.b)("em",{parentName:"li"},"confirmed")," or ",Object(i.b)("em",{parentName:"li"},"discarded"),";"),Object(i.b)("li",{parentName:"ul"},"based on false-discovery rate, it classifies ",Object(i.b)("em",{parentName:"li"},"confirmed")," ERs as\n",Object(i.b)("em",{parentName:"li"},"true-positive")," or ",Object(i.b)("em",{parentName:"li"},"false-positive"),".")),Object(i.b)("p",null,"The following figure is a schematic view of this procedure."),Object(i.b)("p",null,Object(i.b)("img",{alt:"Sets",src:n(107).default})),Object(i.b)("p",null,"(The number of ERs in different sets reported in this figure, are the\nresult of ",Object(i.b)("inlineCode",{parentName:"p"},"wgEncodeSydhTfbsK562CmycStdAlnRep1")," and\n",Object(i.b)("inlineCode",{parentName:"p"},"wgEncodeSydhTfbsK562CmycStdAlnRep2")," (available for download from\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"../sample_data"}),"Sample Data page"),") comparative analysis using\n",Object(i.b)("inlineCode",{parentName:"p"},"-r bio -w 1e-4 -s 1e-8")," parameters.)"))}p.isMDXComponent=!0},95:function(e,t,n){"use strict";n.d(t,"a",(function(){return b})),n.d(t,"b",(function(){return m}));var a=n(0),r=n.n(a);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function s(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function c(e,t){if(null==e)return{};var n,a,r=function(e,t){if(null==e)return{};var n,a,r={},i=Object.keys(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||(r[n]=e[n]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(r[n]=e[n])}return r}var l=r.a.createContext({}),p=function(e){var t=r.a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):s(s({},t),e)),n},b=function(e){var t=p(e.components);return r.a.createElement(l.Provider,{value:t},e.children)},d={inlineCode:"code",wrapper:function(e){var t=e.children;return r.a.createElement(r.a.Fragment,{},t)}},u=r.a.forwardRef((function(e,t){var n=e.components,a=e.mdxType,i=e.originalType,o=e.parentName,l=c(e,["components","mdxType","originalType","parentName"]),b=p(n),u=a,m=b["".concat(o,".").concat(u)]||b[u]||d[u]||i;return n?r.a.createElement(m,s(s({ref:t},l),{},{components:n})):r.a.createElement(m,s({ref:t},l))}));function m(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var i=n.length,o=new Array(i);o[0]=u;var s={};for(var c in t)hasOwnProperty.call(t,c)&&(s[c]=t[c]);s.originalType=e,s.mdxType="string"==typeof e?e:a,o[1]=s;for(var l=2;l<i;l++)o[l]=n[l];return r.a.createElement.apply(null,o)}return r.a.createElement.apply(null,n)}u.displayName="MDXCreateElement"}}]);