(window.webpackJsonp=window.webpackJsonp||[]).push([[17],{89:function(e,t,n){"use strict";n.r(t),n.d(t,"frontMatter",(function(){return o})),n.d(t,"metadata",(function(){return s})),n.d(t,"rightToc",(function(){return c})),n.d(t,"default",(function(){return p}));var a=n(3),r=n(7),i=(n(0),n(96)),o={title:"Welcome",slug:"/"},s={unversionedId:"welcome",id:"welcome",isDocsHomePage:!1,title:"Welcome",description:"The analysis of ChIP-seq samples outputs a number",source:"@site/docs/welcome.md",slug:"/",permalink:"/MSPC/docs/",editUrl:"https://github.com/Genometric/MSPC/tree/dev/website/docs/welcome.md",version:"current",sidebar:"someSidebar",next:{title:"Quick Start",permalink:"/MSPC/docs/quick_start"}},c=[],l={rightToc:c};function p(e){var t=e.components,n=Object(r.a)(e,["components"]);return Object(i.b)("wrapper",Object(a.a)({},l,n,{components:t,mdxType:"MDXLayout"}),Object(i.b)("p",null,'The analysis of ChIP-seq samples outputs a number\nof enriched regions (commonly known as "peaks"),\neach indicating a protein-DNA interaction or a\nspecific chromatin modification. When replicate\nsamples are analyzed, overlapping peaks are expected.\nThis repeated evidence can therefore be used to\nlocally lower the minimum significance required to\naccept a peak. Here, we propose a method for joint\nanalysis of weak peaks. Given a set of peaks from\n(biological or technical) replicates, the method\ncombines the p-values of overlapping enriched regions,\nand allows the "rescue" of weak peaks occurring in\nmore than one replicate.'),Object(i.b)("p",null,"MSPC comparatively evaluates ChIP-seq peaks and\ncombines the statistical significance of\nrepeated evidences, with the aim of lowering the\nminimum significance required to \u201crescue\u201d\nweak peaks; hence reducing false-negatives. "),Object(i.b)("p",null,"MSPC can be used from: "),Object(i.b)("ul",null,Object(i.b)("li",{parentName:"ul"},Object(i.b)("p",{parentName:"li"},Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"/MSPC/docs/cli/about"}),Object(i.b)("strong",{parentName:"a"},"Command Line Interface (CLI)")),": call MSPC CLI from your favorite terminal with necessary\narguments, and it writes the analysis results to BED files.")),Object(i.b)("li",{parentName:"ul"},Object(i.b)("p",{parentName:"li"},Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"/MSPC/docs/library/install"}),Object(i.b)("strong",{parentName:"a"},"As a library in your project")),": call it from your program, and it returns analysis\nresults to your program."))),Object(i.b)("h1",{id:"about"},"About"),Object(i.b)("p",null,"With most peak callers (e.g, MACS), the false-positive\nrate is a function of a user-defined p-value threshold,\nwhere the more conservative thresholds result in lower\nfalse-positive rates---the penalty of which is the\nincrease in the number of false-negatives. While several\nprobabilistic methods are developed to jointly model\nbinding affinities across replicated samples to identify\ncombinatorial enrichment patterns (e.g.,\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38"}),"jMOSAiCS"),",\nor ",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://academic.oup.com/biostatistics/article/15/2/296/226404"}),"this"),",\nor ",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14"}),"this"),",\nor ",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full"}),"this"),"),\nthe more commonly used peak callers (e.g., MACS) operate\non single samples. Additionally, the choice of peak\ncaller is generally an established step in a ChIP-seq\nanalysis pipeline, since altering it may require changes\nto the data pre- and post-processing of the pipeline.\nTherefore, the remainder of this manuscript is focused\non the post peak calling methods to lower false-positive\nrates as they can be applied on the output of any peak\ncaller with minimal changes to the established analysis\npipeline, and can combine evidence across replicated samples."),Object(i.b)("h1",{id:"motivation"},"Motivation"),Object(i.b)("p",null,"Ultimately, with most peak callers (e.g, MACS), the false-positive\nrate is a function of a user-defined p-value threshold, where the more\nconservative thresholds result in lower false-positive rates---the penalty\nof which is the increase in the number of false-negatives. While several\nprobabilistic methods are developed to jointly model binding affinities\nacross replicated samples to identify combinatorial enrichment patterns\n(e.g.,\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38"}),"REF1"),",\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://academic.oup.com/biostatistics/article/15/2/296/226404"}),"REF2"),",\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14"}),"REF3"),",\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://academic.oup.com/bioinformatics/article/31/1/17/2366199"}),"REF4"),",\n",Object(i.b)("a",Object(a.a)({parentName:"p"},{href:"https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full"}),"REF5"),"),\nthe more commonly used peak callers\n(e.g., MACS) operate on single samples. Additionally, the choice of peak\ncaller is generally an established step in a ChIP-seq analysis pipeline,\nsince altering it may require changes to the data pre- and post-processing\nof the pipeline."))}p.isMDXComponent=!0},96:function(e,t,n){"use strict";n.d(t,"a",(function(){return m})),n.d(t,"b",(function(){return h}));var a=n(0),r=n.n(a);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function s(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function c(e,t){if(null==e)return{};var n,a,r=function(e,t){if(null==e)return{};var n,a,r={},i=Object.keys(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||(r[n]=e[n]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(r[n]=e[n])}return r}var l=r.a.createContext({}),p=function(e){var t=r.a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):s(s({},t),e)),n},m=function(e){var t=p(e.components);return r.a.createElement(l.Provider,{value:t},e.children)},b={inlineCode:"code",wrapper:function(e){var t=e.children;return r.a.createElement(r.a.Fragment,{},t)}},u=r.a.forwardRef((function(e,t){var n=e.components,a=e.mdxType,i=e.originalType,o=e.parentName,l=c(e,["components","mdxType","originalType","parentName"]),m=p(n),u=a,h=m["".concat(o,".").concat(u)]||m[u]||b[u]||i;return n?r.a.createElement(h,s(s({ref:t},l),{},{components:n})):r.a.createElement(h,s({ref:t},l))}));function h(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var i=n.length,o=new Array(i);o[0]=u;var s={};for(var c in t)hasOwnProperty.call(t,c)&&(s[c]=t[c]);s.originalType=e,s.mdxType="string"==typeof e?e:a,o[1]=s;for(var l=2;l<i;l++)o[l]=n[l];return r.a.createElement.apply(null,o)}return r.a.createElement.apply(null,n)}u.displayName="MDXCreateElement"}}]);