<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>About · MSPC</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="The analysis of ChIP-seq samples outputs a number of enriched regions "/><meta name="docsearch:language" content="en"/><meta property="og:title" content="About · MSPC"/><meta property="og:type" content="website"/><meta property="og:url" content="https://genometric.github.io/MSPC/index.html"/><meta property="og:description" content="The analysis of ChIP-seq samples outputs a number of enriched regions "/><meta name="twitter:card" content="summary"/><link rel="shortcut icon" href="/MSPC/img/favicon.png"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/vs2015.min.css"/><script>
              (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
              (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
              m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
              })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

              ga('create', 'UA-100863841-2', 'auto');
              ga('send', 'pageview');
            </script><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><script type="text/javascript" src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js" async=""></script><script type="text/javascript" src="https://badge.dimensions.ai/badge.js" async=""></script><link rel="stylesheet" href="/MSPC/css/main.css"/></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/MSPC/"><img class="logo" src="/MSPC/img/mspc_logo.svg" alt="MSPC"/><h2 class="headerTitleWithLogo">MSPC</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class="siteNavGroupActive"><a href="/MSPC/docs/welcome" target="_self">Documentation</a></li><li class=""><a href="https://github.com/Genometric/MSPC/" target="_self">Source Code</a></li><li class=""><a href="https://github.com/Genometric/MSPC/releases" target="_self">Download</a></li><li class=""><a href="https://github.com/Genometric/MSPC/issues" target="_self">Questions</a></li><li class=""><a href="/MSPC/publications" target="_self">Publications</a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="container docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><i></i></div><h2><i>›</i><span>Method</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle">Getting Started</h3><ul><li class="navListItem"><a class="navItem" href="/MSPC/docs/welcome">Welcome</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/quick_start">Quick Start</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/installation">Instalation</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/sample_data">Sample Data</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Method</h3><ul><li class="navListItem navListItemActive"><a class="navItem" href="/MSPC/docs/method/about">About</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/method/sets">Sets</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/method/consensus">Consensus peaks</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">CLI</h3><ul><li class="navListItem"><a class="navItem" href="/MSPC/docs/cli/about">About</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/cli/input">Input</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/cli/output">Output</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/cli/args">Arguments</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/cli/parser">Parser Configuration</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Library</h3><ul><li class="navListItem"><a class="navItem" href="/MSPC/docs/library/install">Install</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/library/basic_invocation">Basic Invocation</a></li><li class="navListItem"><a class="navItem" href="/MSPC/docs/library/asynch_exe">(A)synchronous execution</a></li></ul></div></div></section></div><script>
            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              const headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                if (event.target.tagName === 'A') {
                  document.body.classList.remove('tocActive');
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer docMainContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 class="postHeaderTitle">About</h1></header><article><div><span><p>The analysis of ChIP-seq samples outputs a number of enriched regions
(commonly known as &quot;peaks&quot;), each indicating a protein-DNA interaction
or a specific chromatin modification.</p>
<p>When replicate samples are analyzed, overlapping peaks are expected.
This repeated evidence can therefore be used to locally lower the
minimum significance required to accept a peak. MSPC is a method for
joint analysis of peaks.</p>
<p>Given a set of peaks from (biological or technical) replicates,
MSPC combines the p-values of overlapping enriched regions, and allows
the &quot;rescue&quot; of weak peaks occurring in more than one replicate.</p>
<h2><a class="anchor" aria-hidden="true" id="remarks"></a><a href="#remarks" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Remarks</h2>
<p>MSPC rigorously combines the statistical significance of the overlapping
enriched regions (ER), in order to &quot;rescue&quot; weak peaks, which would probably
be discarded in a single-sample analysis, when their combined evidence
across multiple samples is sufficiently strong.
[<a href="https://doi.org/10.1093/bioinformatics/btv293">Ref</a>]</p>
<p>MSPC takes as input, for each replicate, a list of ERs and a measure of
their individual significance in terms of a p-value.
Starting from a permissive call, it divides the initial ERs in &quot;stringent&quot;
(highly significant) and &quot;weak&quot; (moderately significant), and it assesses
the presence of overlapping enriched regions across multiple replicates.
Non-overlapping regions can be penalized or discarded according to specific
needs.</p>
<p>The significance of the overlapping regions is rigorously combined with the
<a href="https://en.wikipedia.org/wiki/Fisher%27s_method">Fisher's method</a> to obtain
a global score. This score is assessed against an adjustable threshold on
the combined evidence, and peaks in each replicate are either confirmed or
discarded.</p>
<p>Finally, in order to account for multiple testing correction, MSPC applies the
<a href="https://en.wikipedia.org/wiki/False_discovery_rate#Benjamini�Hochberg_procedure">Benjamini-Hochberg procedure</a>,
and outputs ERs with false-discovery rate smaller than an adjustable threshold.</p>
<p>This flow is captured in the following flowchart for each ER of each replicate:</p>
<p><img src="/MSPC/docs/assets/simplified_flow_chart.svg" alt="alt text"></p>
<p>(This flowchart is a simplified version of the flowchart available in
<a href="https://doi.org/10.1093/bioinformatics/btv293">MSPC's manuscript</a>.)</p>
<p>In other words:</p>
<ul>
<li>for each replicate, MSPC classifies input ERs as <em>stringent</em>, <em>weak</em>,
and <em>background</em>, based on their p-value;</li>
<li>performs a comparative analysis, and based on the combined stringency test,
it classifies <em>stringent</em> and <em>weak</em> ERs as <em>confirmed</em> or <em>discarded</em>;</li>
<li>based on false-discovery rate, it classifies <em>confirmed</em> ERs as
<em>true-positive</em> or <em>false-positive</em>.</li>
</ul>
<p>The following figure is a schematic view of this procedure.</p>
<p><img src="/MSPC/docs/assets/sets.svg" alt="alt text"></p>
<p>(The number of ERs in different sets reported in this figure, are the
result of <code>wgEncodeSydhTfbsK562CmycStdAlnRep1</code> and
<code>wgEncodeSydhTfbsK562CmycStdAlnRep2</code> (available for download from
<a href="sample_data">Sample Data page</a>) comparative analysis using
<code>-r bio -w 1e-4 -s 1e-8</code> parameters.)</p>
</span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/MSPC/docs/sample_data"><span class="arrow-prev">← </span><span>Previous</span></a><a class="docs-next button" href="/MSPC/docs/method/sets"><span>Next</span><span class="arrow-next"> →</span></a></div><p style="font-size:12px;text-align:right"><strong>Last updated: </strong>2018-12-22 18:25:34</p></div></div><nav class="onPageNav docOnPageNav"><ul class="toc-headings"><li><a href="#remarks">Remarks</a></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="sitemap"><a href="/MSPC/" class="nav-home"><img src="/MSPC/img/mspc_logo.svg" alt="MSPC" width="66" height="58"/></a><div><h5>Docs</h5><a href="/MSPC/docs/en/quick_start.html">Quick Start</a></div><div><h5>Community</h5><a href="https://github.com/Genometric/MSPC">GitHub</a></div></section><section class="copyright">Copyright © 2019 Genometric</section></footer></div></body></html>