<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.69">
<link rel="search" type="application/opensearchdescription+xml" title="MSPC" href="/MSPC/opensearch.xml"><title data-react-helmet="true">Welcome | MSPC</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Welcome | MSPC"><meta data-react-helmet="true" name="description" content="The analysis of ChIP-seq samples outputs a number"><meta data-react-helmet="true" property="og:description" content="The analysis of ChIP-seq samples outputs a number"><meta data-react-helmet="true" property="og:url" content="https://genometric.github.io/MSPC/docs/"><link data-react-helmet="true" rel="shortcut icon" href="/MSPC/logo/logo.svg"><link data-react-helmet="true" rel="preconnect" href="https://7KY7XPQTMJ-dsn.algolia.net" crossorigin="anonymous"><link data-react-helmet="true" rel="canonical" href="https://genometric.github.io/MSPC/docs/"><link rel="stylesheet" href="/MSPC/styles.81b4d860.css">
<link rel="preload" href="/MSPC/styles.ddc85fd4.js" as="script">
<link rel="preload" href="/MSPC/runtime~main.f0134c96.js" as="script">
<link rel="preload" href="/MSPC/main.d509c102.js" as="script">
<link rel="preload" href="/MSPC/1.65bea949.js" as="script">
<link rel="preload" href="/MSPC/24.aa3df51f.js" as="script">
<link rel="preload" href="/MSPC/27.e683cd83.js" as="script">
<link rel="preload" href="/MSPC/935f2afb.b09b5a54.js" as="script">
<link rel="preload" href="/MSPC/23.535c49b7.js" as="script">
<link rel="preload" href="/MSPC/ebfee794.96311fde.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_2AhQ">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/MSPC/"><img src="/MSPC/logo/logo.svg" alt="logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img src="/MSPC/logo/logo.svg" alt="logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">MSPC</strong></a></div><div class="navbar__items navbar__items--right"><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/MSPC/docs/">Documentation</a><a href="https://github.com/Genometric/MSPC/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Source Code</a><a href="https://github.com/Genometric/MSPC/releases" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Download</a><a href="https://github.com/Genometric/MSPC/issues" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Questions</a><a class="navbar__item navbar__link" href="/MSPC/Publications">Publications</a><div class="react-toggle react-toggle--disabled displayOnlyInLargeViewport_2aTZ"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_BsTx">üåú</span></div><div class="react-toggle-track-x"><span class="toggle_BsTx">üåû</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" disabled="" aria-label="Dark mode toggle" class="react-toggle-screenreader-only"></div><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span><span class="DocSearch-Button-Key">‚åò</span><span class="DocSearch-Button-Key">K</span></button></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/MSPC/"><img src="/MSPC/logo/logo.svg" alt="logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img src="/MSPC/logo/logo.svg" alt="logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">MSPC</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/MSPC/docs/">Documentation</a></li><li class="menu__list-item"><a href="https://github.com/Genometric/MSPC/" target="_blank" rel="noopener noreferrer" class="menu__link">Source Code</a></li><li class="menu__list-item"><a href="https://github.com/Genometric/MSPC/releases" target="_blank" rel="noopener noreferrer" class="menu__link">Download</a></li><li class="menu__list-item"><a href="https://github.com/Genometric/MSPC/issues" target="_blank" rel="noopener noreferrer" class="menu__link">Questions</a></li><li class="menu__list-item"><a class="menu__link" href="/MSPC/Publications">Publications</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_2gpo"><div class="docSidebarContainer_3_JD" role="complementary"><div class="sidebar_2urC"><div class="menu menu--responsive thin-scrollbar menu_5FrY"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_Dm3K" xmlns="http://www.w3.org/2000/svg" height="24" width="24" viewBox="0 0 32 32" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Getting Started</a><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/MSPC/docs/">Welcome</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/MSPC/docs/quick_start">Quick Start</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/MSPC/docs/installation">Installation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/MSPC/docs/sample_data">Sample Data</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Method</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/method/about">About</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/method/sets">Sets</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/method/consensus">Consensus peaks</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">CLI</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/cli/about">About</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/cli/input">Input</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/cli/output">Output</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/cli/args">Arguments</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/cli/parser">Parser Configuration</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Library</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/library/install">Install</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/library/basic_invocation">Basic Invocation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/MSPC/docs/library/asynch_exe">(A)synchronous execution</a></li></ul></li></ul></div></div></div><main class="docMainContainer_3EyW"><div class="container padding-vert--lg docItemWrapper_39qw"><div class="row"><div class="col docItemCol_2ASc"><div class="docItemContainer_3QWW"><article><header><h1 class="docTitle_1Lrw">Welcome</h1></header><div class="markdown"><p>The analysis of ChIP-seq samples outputs a number
of enriched regions (commonly known as &quot;peaks&quot;),
each indicating a protein-DNA interaction or a
specific chromatin modification. When replicate
samples are analyzed, overlapping peaks are expected.
This repeated evidence can therefore be used to
locally lower the minimum significance required to
accept a peak. Here, we propose a method for joint
analysis of weak peaks. Given a set of peaks from
(biological or technical) replicates, the method
combines the p-values of overlapping enriched regions,
and allows the &quot;rescue&quot; of weak peaks occurring in
more than one replicate.</p><p>MSPC comparatively evaluates ChIP-seq peaks and
combines the statistical significance of
repeated evidences, with the aim of lowering the
minimum significance required to ‚Äúrescue‚Äù
weak peaks; hence reducing false-negatives. </p><p>MSPC can be used from: </p><ul><li><p><a href="/MSPC/docs/cli/about"><strong>Command Line Interface (CLI)</strong></a>: call MSPC CLI from your favorite terminal with necessary
arguments, and it writes the analysis results to BED files.</p></li><li><p><a href="/MSPC/docs/library/install"><strong>As a library in your project</strong></a>: call it from your program, and it returns analysis
results to your program.</p></li></ul><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="about"></a>About<a class="hash-link" href="#about" title="Direct link to heading">#</a></h1><p>With most peak callers (e.g, MACS), the false-positive
rate is a function of a user-defined p-value threshold,
where the more conservative thresholds result in lower
false-positive rates---the penalty of which is the
increase in the number of false-negatives. While several
probabilistic methods are developed to jointly model
binding affinities across replicated samples to identify
combinatorial enrichment patterns (e.g.,
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38" target="_blank" rel="noopener noreferrer">jMOSAiCS</a>,
or <a href="https://academic.oup.com/biostatistics/article/15/2/296/226404" target="_blank" rel="noopener noreferrer">this</a>,
or <a href="https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14" target="_blank" rel="noopener noreferrer">this</a>,
or <a href="https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full" target="_blank" rel="noopener noreferrer">this</a>),
the more commonly used peak callers (e.g., MACS) operate
on single samples. Additionally, the choice of peak
caller is generally an established step in a ChIP-seq
analysis pipeline, since altering it may require changes
to the data pre- and post-processing of the pipeline.
Therefore, the remainder of this manuscript is focused
on the post peak calling methods to lower false-positive
rates as they can be applied on the output of any peak
caller with minimal changes to the established analysis
pipeline, and can combine evidence across replicated samples.</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="motivation"></a>Motivation<a class="hash-link" href="#motivation" title="Direct link to heading">#</a></h1><p>Ultimately, with most peak callers (e.g, MACS), the false-positive
rate is a function of a user-defined p-value threshold, where the more
conservative thresholds result in lower false-positive rates---the penalty
of which is the increase in the number of false-negatives. While several
probabilistic methods are developed to jointly model binding affinities
across replicated samples to identify combinatorial enrichment patterns
(e.g.,
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38" target="_blank" rel="noopener noreferrer">REF1</a>,
<a href="https://academic.oup.com/biostatistics/article/15/2/296/226404" target="_blank" rel="noopener noreferrer">REF2</a>,
<a href="https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14" target="_blank" rel="noopener noreferrer">REF3</a>,
<a href="https://academic.oup.com/bioinformatics/article/31/1/17/2366199" target="_blank" rel="noopener noreferrer">REF4</a>,
<a href="https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full" target="_blank" rel="noopener noreferrer">REF5</a>),
the more commonly used peak callers
(e.g., MACS) operate on single samples. Additionally, the choice of peak
caller is generally an established step in a ChIP-seq analysis pipeline,
since altering it may require changes to the data pre- and post-processing
of the pipeline.</p></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/Genometric/MSPC/tree/dev/website/docs/welcome.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 40 40" style="margin-right:0.3em;vertical-align:sub"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/MSPC/docs/quick_start"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Quick Start ¬ª</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_3SO_ thin-scrollbar"></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/MSPC/docs/quick_start/">Quick Start</a></li><li class="footer__item"><a class="footer__link-item" href="/MSPC/docs/installation/">Installation</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Community</h4><ul class="footer__items"><li class="footer__item"><a href="https://github.com/Genometric/MSPC" target="_blank" rel="noopener noreferrer" class="footer__link-item">Github</a></li></ul></div></div><div class="text--center"><div>Copyright ¬© 2021 Genometric</div></div></div></footer></div>
<script src="/MSPC/styles.ddc85fd4.js"></script>
<script src="/MSPC/runtime~main.f0134c96.js"></script>
<script src="/MSPC/main.d509c102.js"></script>
<script src="/MSPC/1.65bea949.js"></script>
<script src="/MSPC/24.aa3df51f.js"></script>
<script src="/MSPC/27.e683cd83.js"></script>
<script src="/MSPC/935f2afb.b09b5a54.js"></script>
<script src="/MSPC/23.535c49b7.js"></script>
<script src="/MSPC/ebfee794.96311fde.js"></script>
</body>
</html>