<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-welcome">
<head>
<meta charset="UTF-8">
<meta name="generator" content="Docusaurus v2.1.0">
<title data-rh="true">Welcome | MSPC</title><meta data-rh="true" name="viewport" content="width=device-width,initial-scale=1"><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://genometric.github.io/MSPC/docs/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Welcome | MSPC"><meta data-rh="true" name="description" content="The analysis of ChIP-seq samples outputs a number"><meta data-rh="true" property="og:description" content="The analysis of ChIP-seq samples outputs a number"><link data-rh="true" rel="icon" href="/MSPC/logo/logo.svg"><link data-rh="true" rel="canonical" href="https://genometric.github.io/MSPC/docs/"><link data-rh="true" rel="alternate" href="https://genometric.github.io/MSPC/docs/" hreflang="en"><link data-rh="true" rel="alternate" href="https://genometric.github.io/MSPC/docs/" hreflang="x-default"><link data-rh="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="search" type="application/opensearchdescription+xml" title="MSPC" href="/MSPC/opensearch.xml"><link rel="stylesheet" href="/MSPC/assets/css/styles.b6cc29ab.css">
<link rel="preload" href="/MSPC/assets/js/runtime~main.c2e3fee7.js" as="script">
<link rel="preload" href="/MSPC/assets/js/main.ac441654.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region" aria-label="theme.common.skipToMainContent"><a href="#" class="skipToContent_fXgn">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/MSPC/"><div class="navbar__logo"><img src="/MSPC/logo/logo.svg" alt="logo" class="themedImage_ToTc themedImage--light_HNdA"><img src="/MSPC/logo/logo.svg" alt="logo" class="themedImage_ToTc themedImage--dark_i4oU"></div><b class="navbar__title text--truncate">MSPC</b></a></div><div class="navbar__items navbar__items--right"><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/MSPC/docs/">Documentation</a><a href="https://github.com/Genometric/MSPC/" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Source Code<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><a href="https://github.com/Genometric/MSPC/releases" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Download<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><a href="https://github.com/Genometric/MSPC/issues" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">Questions<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><a class="navbar__item navbar__link" href="/MSPC/Publications">Publications</a><div class="toggle_vylO colorModeToggle_DEke"><button class="clean-btn toggleButton_gllP toggleButtonDisabled_aARS" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_pyhR"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_wfgR"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_ZlJk"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper mainWrapper_z2l0 docsWrapper_BCFX"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_sjWU" type="button"></button><div class="docPage__5DB"><aside class="theme-doc-sidebar-container docSidebarContainer_b6E3"><div class="sidebar_njMd"><nav class="menu thin-scrollbar menu_SIkG"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret menu__link--active" aria-expanded="true" href="/MSPC/docs/">Getting Started</a></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/MSPC/docs/">Welcome</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/MSPC/docs/quick_start">Quick Start</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/MSPC/docs/installation">Installation</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/MSPC/docs/sample_data">Sample Data</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/MSPC/docs/method/about">Method</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/MSPC/docs/cli/about">CLI</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/MSPC/docs/library/install">Library</a></div></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_PEFL"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_kv0_"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_gTbr"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_VOVn"><div class="docItemContainer_Djhp"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Z_bl" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/MSPC/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_OVgt"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li class="breadcrumbs__item"><span class="breadcrumbs__link">Getting Started</span><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Welcome</span><meta itemprop="position" content="2"></li></ul></nav><div class="theme-doc-markdown markdown"><header><h1>Welcome</h1></header><p>The analysis of ChIP-seq samples outputs a number
of enriched regions (commonly known as &quot;peaks&quot;),
each indicating a protein-DNA interaction or a
specific chromatin modification. When replicate
samples are analyzed, overlapping peaks are expected.
This repeated evidence can therefore be used to
locally lower the minimum significance required to
accept a peak. Here, we propose a method for joint
analysis of weak peaks. Given a set of peaks from
(biological or technical) replicates, the method
combines the p-values of overlapping enriched regions,
and allows the &quot;rescue&quot; of weak peaks occurring in
more than one replicate.</p><p>MSPC comparatively evaluates ChIP-seq peaks and
combines the statistical significance of
repeated evidences, with the aim of lowering the
minimum significance required to “rescue”
weak peaks; hence reducing false-negatives. </p><p>MSPC can be used from: </p><ul><li><p><a href="/MSPC/docs/cli/about"><strong>Command Line Interface (CLI)</strong></a>: call MSPC CLI from your favorite terminal with necessary
arguments, and it writes the analysis results to BED files.</p></li><li><p><a href="/MSPC/docs/library/install"><strong>As a library in your project</strong></a>: call it from your program, and it returns analysis
results to your program.</p></li><li><p><a href="https://bioconductor.org/packages/release/bioc/html/rmspc.html" target="_blank" rel="noopener noreferrer"><strong>As an R package from Bioconductor</strong></a>:
The Bioconductor package is a wrapper around the CLI application, hence
offering the same functionality as the CLI application. The package can be
invoked from R to process files or
<a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html" target="_blank" rel="noopener noreferrer">GenomicRanges</a>
(Granges) objects, and the results are stored as Granges objects.
Please refer to <a href="https://bioconductor.org/packages/release/bioc/vignettes/rmspc/inst/doc/rmpsc.html" target="_blank" rel="noopener noreferrer">Bioconductor user guide</a>.</p></li></ul><h1>About</h1><p>With most peak callers (e.g, MACS), the false-positive
rate is a function of a user-defined p-value threshold,
where the more conservative thresholds result in lower
false-positive rates---the penalty of which is the
increase in the number of false-negatives. While several
probabilistic methods are developed to jointly model
binding affinities across replicated samples to identify
combinatorial enrichment patterns (e.g.,
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38" target="_blank" rel="noopener noreferrer">jMOSAiCS</a>,
or <a href="https://academic.oup.com/biostatistics/article/15/2/296/226404" target="_blank" rel="noopener noreferrer">this</a>,
or <a href="https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14" target="_blank" rel="noopener noreferrer">this</a>,
or <a href="https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full" target="_blank" rel="noopener noreferrer">this</a>),
the more commonly used peak callers (e.g., MACS) operate
on single samples. Additionally, the choice of peak
caller is generally an established step in a ChIP-seq
analysis pipeline, since altering it may require changes
to the data pre- and post-processing of the pipeline.
Therefore, the remainder of this manuscript is focused
on the post peak calling methods to lower false-positive
rates as they can be applied on the output of any peak
caller with minimal changes to the established analysis
pipeline, and can combine evidence across replicated samples.</p><h1>Motivation</h1><p>Ultimately, with most peak callers (e.g, MACS), the false-positive
rate is a function of a user-defined p-value threshold, where the more
conservative thresholds result in lower false-positive rates---the penalty
of which is the increase in the number of false-negatives. While several
probabilistic methods are developed to jointly model binding affinities
across replicated samples to identify combinatorial enrichment patterns
(e.g.,
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r38" target="_blank" rel="noopener noreferrer">REF1</a>,
<a href="https://academic.oup.com/biostatistics/article/15/2/296/226404" target="_blank" rel="noopener noreferrer">REF2</a>,
<a href="https://link.springer.com/chapter/10.1007/978-3-319-05269-4_14" target="_blank" rel="noopener noreferrer">REF3</a>,
<a href="https://academic.oup.com/bioinformatics/article/31/1/17/2366199" target="_blank" rel="noopener noreferrer">REF4</a>,
<a href="https://www.frontiersin.org/articles/10.3389/fgene.2018.00731/full" target="_blank" rel="noopener noreferrer">REF5</a>),
the more commonly used peak callers
(e.g., MACS) operate on single samples. Additionally, the choice of peak
caller is generally an established step in a ChIP-seq analysis pipeline,
since altering it may require changes to the data pre- and post-processing
of the pipeline.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/Genometric/MSPC/tree/dev/website/docs/welcome.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_Z9Sw" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_vwxv"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--next" href="/MSPC/docs/quick_start"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Quick Start</div></a></nav></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/MSPC/docs/quick_start/">Quick Start</a></li><li class="footer__item"><a class="footer__link-item" href="/MSPC/docs/installation/">Installation</a></li></ul></div><div class="col footer__col"><div class="footer__title">Community</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/Genometric/MSPC" target="_blank" rel="noopener noreferrer" class="footer__link-item">Github<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_nPIU"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2022 Genometric</div></div></div></footer></div>
<script src="/MSPC/assets/js/runtime~main.c2e3fee7.js"></script>
<script src="/MSPC/assets/js/main.ac441654.js"></script>
</body>
</html>